src/libraries/System.Net.Security/src/System/Net/Security/SslConnectionInfo.Windows.cs

```diff
@@ -17,7 +17,20 @@ internal partial struct SslConnectionInfo
                 alpnContext.ProtoNegoExt == Interop.ApplicationProtocolNegotiationExt.ALPN &&
                 alpnContext.ProtoNegoStatus == Interop.ApplicationProtocolNegotiationStatus.Success)
             {
-                return alpnContext.Protocol;
+                if (alpnContext.ProtocolIdSize == s_http1.Length && alpnContext.Protocol.SequenceEqual(s_http1))
```

Comment:
Why does the Windows one check ProtocolIdSize but the Linux one doesn't?

Reply:
The code originally looked different and I did not want to allocate the Span unless we need to. Perhaps that is cheap. Here I have the size available directly so I left the check in place. On Linux, The Interop function returns always Span so the code is somewhat different. I can remove the extra check if we want to.